packages/date-picker/src/basic/date-table.vue

```diff
@@ -167,26 +167,28 @@
             cell.start = this.minDate && time === getDateTimestamp(this.minDate);
             cell.end = this.maxDate && time === getDateTimestamp(this.maxDate);
             const isToday = time === now;
+            const getFirstDayPosition = () => day + offset < 0 ? 7 + day + offset : i * 7 + j;
 
             if (isToday) {
               cell.type = 'today';
             }
 
             if (i >= 0 && i <= 1) {
-              if (j + i * 7 >= (day + offset)) {
+              const flag = day + offset < 0 ? 7 + day + offset : day + offset;
```

Comment:
what does `flag` mean. It should be meaningful.

Reply:
any suggestions?

Reply:
numberOfDaysFromPreviousMonth ?

Reply:
I have updated to `prevMonthElementsLength `, if not ok, I accept your idea.

Reply:
`number` seems better, it replaces elements & length.

IMO, the word `element` usually refers to DOM element, which is not rendered here.

Reply:
resolved. thanks.